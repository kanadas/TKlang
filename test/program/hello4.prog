stream buffer
input Input
state 
    tail l = match l with {
        h:t -> t;
        [] -> []
        };
    head l = match l with {
        h:t -> 1@h;
        [] -> 2@()
        };
    buf :: [char];
    buf = tail buf
output 
    out = head buf
initial 
    buf = "Hello world!", 
    out = 2 @ ();

stream main
input buffer
state
output print = buffer.out
initial print = 0
