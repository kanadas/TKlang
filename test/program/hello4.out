program/hello4.prog
Stream: buffer has uninitialized values
